projects/libs/core-library-typescript/src/common-library/startup-check-runner.ts

```diff
@@ -0,0 +1,52 @@
+import { StartupCheck } from "./startup-check";
+import { StartupStatus, STARTUP_STATUS } from "./startup-status";
+import { logger } from "../common-library";
+
+export class StartupCheckRunner {
+  name: string;
+  startupCheckers: StartupCheck[] = [];
+
+  constructor(name: string, startupCheckers: StartupCheck[]) {
+    this.name = name;
+    this.startupCheckers = startupCheckers;
+  }
+
+  run(): Promise<StartupStatus> {
+
+    const startupCheckPromises = this.startupCheckers.map((s: StartupCheck) => s.startupCheck());
+    return Promise.all(startupCheckPromises)
+      .then(this.mergeStartupStatuses)
+      .catch(err => {
+        logger.error(err);
+        return <StartupStatus>{ status: STARTUP_STATUS.FAIL, message: err.message };
+      });
+  }
+
+  private mergeStartupStatuses = (startupStatuses: StartupStatus[]): StartupStatus => {
+
+    let hasAtLeastOneOk = false;
+    let hasAtLeastOneFailure = false;
+
+    for (const item of startupStatuses) {
+      if (item.status === STARTUP_STATUS.OK) {
+        hasAtLeastOneOk = true;
+      } else {
+        hasAtLeastOneFailure = true;
+      }
+    }
+
+    if (!hasAtLeastOneFailure) {
+      return <StartupStatus>{ name: this.name, status: STARTUP_STATUS.OK, childStartupStatuses: startupStatuses };
+    }
+    if (hasAtLeastOneOk) {
+      return <StartupStatus>{ name: this.name, status: STARTUP_STATUS.PARTIALLY_OPERATIONAL, childStartupStatuses: startupStatuses };
+    }
+    return <StartupStatus>{ name: this.name, status: STARTUP_STATUS.FAIL, childStartupStatuses: startupStatuses };
+
```

Comment:
extract 'status' as a variable named 'statusToReturn' or 'mergedStatus' and make a sign return statement - it will be more readable, could be that you can let go of the boolean flags altogether then.